src/etos_test_runner/lib/testrunner.py

```diff
@@ -124,7 +149,22 @@ def outcome(self, result, executed, description):
         :return: Outcome of test execution.
         :rtype: dict
         """
-        if executed:
+        test_framework_output = {
+            "test_framework_exit_codes": test_framework_exit_codes,
+        }
+        custom_verdict = self.verdict_matcher.evaluate(test_framework_output)
+        if custom_verdict is not None:
+            try:
+                conclusion = custom_verdict["conclusion"]
+                verdict = custom_verdict["verdict"]
+                description = custom_verdict["description"]
+            except KeyError as err:
+                raise ValueError(
+                    f"Malformed entry in the verdict rule file: {custom_verdict}. "
+                    "Expected attributes: description, condition, conclusion, verdict."
+                ) from err
+            self.logger.info("Verdict matches testrunner verdict rule: %s", custom_verdict)
+        elif executed:
```

Comment:
Shouldn't this check be done before we evaluate exit codes? The verdict_matcher will run on an empty list of exit codes if `executed` is not True.